src/agent.js

```diff
@@ -7,7 +8,13 @@ const handleTransaction = async (txEvent) => {
     deployerWatchAgent.handleTransaction(txEvent),
   ]);
 
+  const [monitorMintCallsFindings] = await Promise.all([
+    monitorMintCallsAgent.handleTransaction(txEvent),
+  ]);
+
   findings.push(...deployerWatchFindings);
+  findings.push(...monitorMintCallsFindings);
+
   return findings;
```

Comment:
this should not be in a separate `Promise.all`. it should be in the same guy as `deployerWatchAgent`

I suggest you remove the array capturing thing with pushing the spread arrays and just use return the flat result

```
> a = [[1],[2],[3]]
[ [ 1 ], [ 2 ], [ 3 ] ]
> a.flat()
[ 1, 2, 3 ]
```

```
  const findings = (
    await Promise.all([
      deployerWatchAgent.handleTransaction(txEvent),
      monitorMintCallsAgent.handleTransaction(txEvent),
    ]);
  ).flat()```

or something along the lines of that...

Reply:
Good catch.  I consolidated the promises into one Promise.all() the way you suggested (it appeared to function appropriately in `npm test` and `npm start`).